### PR TITLE
[mqtt.homeassistant] Bugfix in HomeAssistantThingHandler

### PR DESCRIPTION
**Pull request by mehoekstra**
Hello,

Great work on the mqtt bindings @davidgraeff, @jochen314 and others.  While trying to get a custom RGB light working with MQTT/HomeAssistant I encountered a few issues.  I submitted separate pull requests for each.

For this one:
Availability status was not updating correctly.  I debugged this in the code from a few days ago and discovered that the subscription never happened for the availability channel.  I fixed that, but then discovered that the availability services were changed significantly over the last day.  The updated code still had one bug.  In HomeAssistantThing handler the if statement said to set the Thing to ONLINE if the availability message was received, or if the availability topic was online.  But this had the effect of always being ONLINE since the OR matched message received after the first update, even if an OFFLINE message were sent.  I believe you meant it should be ONLINE if no message has been received, or if the ONLINE message has been received.  This change fixed the behavior for me.

### CODE REVIEW

**Comment by mehoekstra** [comment]
Not sure what these errors are about, as you can see I made very few changes from the master branch (changed 2 lines of code).  I will look at them them tomorrow.  Any ideas @jcohen314?

**Comment by mehoekstra** [comment]
Not sure why Travis build failed, trying again.

**Comment by mehoekstra** [comment]
@davidgraeff and @jochen314, I'm trying to be helpful by submitting my small bugfixes, but the heck if I can figure out what's wrong with the Travis CI build.  It appears broken on main the 2.5.x branch independent of my changes.  The Jenkins builds seem to work.  Please advise if you'd like to see me do anything else.

**Comment by mehoekstra** [comment]
See related conversation in the openhab community discussion forum here.  https://community.openhab.org/t/travis-ci-build-for-org-openhab-binding-mqtt-homeassistant/93042

File: bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java [hunk 1]
```diff
@@ -297,7 +297,7 @@ public void accept(List<AbstractComponent<?>> discoveredComponentsList) {
 
     @Override
     protected void updateThingStatus(boolean messageReceived, boolean availabilityTopicsSeen) {
-        if (messageReceived || availabilityTopicsSeen) {
+        if (!messageReceived || availabilityTopicsSeen) {
```
Comment (9037568) [inline bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java]: This is incorrect and needs to be reverted.

Comment (mehoekstra) [reply bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/handler/HomeAssistantThingHandler.java]: There has been an ongoing discussion on this here:   https://community.openhab.org/t/travis-ci-build-for-org-openhab-binding-mqtt-homeassistant/93042, maybe the wrong place to have it.  With the current logic, without the change, at least an MQTT Homeassistant light will never go offline when the payload of the MQTT availability_topic changes to offline.  Perhaps the proper fix belongs someplace else.  Thoughts?

**Comment by mehoekstra** [comment]
Making this change request just about the change in logic to HomeAssistantThingHander since it's generated the most discussion so far.  Will submit separate pull requests for the other two items.

**Comment by mehoekstra** [comment]
Darn it, I completely f'ed up this branch by trying to go back and sign intermediate commits that I forgot to sign.  Unfortunately I'm going to have to close and start over.  I'll reference this PR for the discussion.

**Pull request closed without merging**